git_machete/cmd.py

```diff
@@ -1347,22 +1340,22 @@ def log(self, branch: str) -> None:
         run_git(self.cli_ctxt, "log", "^" + self.fork_point(branch, use_overrides=True), f"refs/heads/{branch}")
 
 
-def slidable(machete_context: MacheteClient) -> List[str]:
-    return [b for b in machete_context.managed_branches if b in machete_context.up_branch]
+def slidable(machete_client: MacheteClient) -> List[str]:
```

Comment:
Hmmm actually... now that `MacheteClient` has a wider responsibility (and no longer corresponds just to the CLI/GUI-independent parts of the logic)... I think methods like `slidable`, `down` and the similar ones can also go into `MacheteClient`.

Reply:
The things I would rather still NOT see in `MacheteClient` are:

1. Utilities that can be used in other pieces of software that know of `git` but don't know of `git-machete` (e.g. `get_git_version`) or the ones that don't even know of `git` (e.g. `directory_exists`) ➡️ they should go into separate Python subpackages
2. Any method that related to parsing command-line params/input - `MacheteClient` should be suitable for **programmatic access**, i.e. directly from Python code rather than from command-line

LMK if you've got any questions

Reply:
What about methods, which are connected to fork point, egz. get_fork_point_override_data?
Also sync_annotations_to_github_prs looks like Machete specific.
@PawelLipski What do you think?